pkg/sources/source.go

```diff
@@ -35,3 +38,96 @@ type Source interface {
 	RoutesNotExported(ctx context.Context, neighborID string) (*api.RoutesResponse, error)
 	AllRoutes(context.Context) (*api.RoutesResponse, error)
 }
+
+func hiddenNeighborsExcludeLists(hiddenNeighbors []string) ([]*net.IPNet, []net.IP, []*regexp.Regexp, error) {
+	excludeCIDRs := make([]*net.IPNet, 0, len(hiddenNeighbors))
+	excludeIPs := make([]net.IP, 0, len(hiddenNeighbors))
+	excludePatterns := make([]*regexp.Regexp, 0, len(hiddenNeighbors))
+	for _, hiddenNeighbor := range hiddenNeighbors {
+		if _, hiddenNet, err := net.ParseCIDR(hiddenNeighbor); err == nil {
+			excludeCIDRs = append(excludeCIDRs, hiddenNet)
+		} else if ip := net.ParseIP(hiddenNeighbor); ip != nil {
+			excludeIPs = append(excludeIPs, ip)
+		} else {
+			pattern, err := regexp.Compile(hiddenNeighbor)
```

Comment:
Another reason to do this at config parse time:
If there is a broken pattern in the config, this would fail when routes refresh starts or, if there is no global search enabled, on first request.